x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/crawler/crawler_logic.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { Actions } from '../../../../shared/api_logic/create_api_logic';
+import { flashAPIErrors, flashSuccessToast } from '../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../shared/http';
+import { GetCrawlerApiLogic, GetCrawlerArgs } from '../../../api/crawler/get_crawler_api_logic';
+import {
+  CrawlerData,
+  CrawlerDomain,
+  CrawlerStatus,
+  CrawlEvent,
+  CrawlRequest,
+} from '../../../api/crawler/types';
+import { IndexNameLogic } from '../index_name_logic';
+
+const POLLING_DURATION = 1000;
+const POLLING_DURATION_ON_FAILURE = 5000;
+const ACTIVE_STATUSES = [
+  CrawlerStatus.Pending,
+  CrawlerStatus.Starting,
+  CrawlerStatus.Running,
+  CrawlerStatus.Canceling,
+];
+
+export interface CrawlRequestOverrides {
+  domain_allowlist?: string[];
+  max_crawl_depth?: number;
+  seed_urls?: string[];
+  sitemap_discovery_disabled?: boolean;
+  sitemap_urls?: string[];
+}
+
+export interface CrawlerValues {
+  dataLoading: boolean;
+  domains: CrawlerDomain[];
+  events: CrawlEvent[];
+  mostRecentCrawlRequest: CrawlRequest | null;
+  mostRecentCrawlRequestStatus: CrawlerStatus | null;
+  timeoutId: NodeJS.Timeout | null;
+}
+
+export type CrawlerActions = Pick<
+  Actions<GetCrawlerArgs, CrawlerData>,
+  'apiError' | 'apiSuccess'
+> & {
+  clearTimeoutId(): void;
+  createNewTimeoutForCrawlerData(duration: number): { duration: number };
+  fetchCrawlerData(): void;
+  onCreateNewTimeout(timeoutId: NodeJS.Timeout): { timeoutId: NodeJS.Timeout };
+  reApplyCrawlRules(domain?: CrawlerDomain): { domain?: CrawlerDomain };
+  startCrawl(overrides?: CrawlRequestOverrides): { overrides?: CrawlRequestOverrides };
+  stopCrawl(): void;
+};
+
+export const CrawlerLogic = kea<MakeLogicType<CrawlerValues, CrawlerActions>>({
+  path: ['enterprise_search', 'crawler_logic'],
+  connect: {
+    // actions: [GetCrawlerApiLogic, ['apiError', 'apiSuccess']],
+    // values: [GetCrawlerApiLogic, ['status']],
+  },
+  actions: {
+    clearTimeoutId: true,
+    createNewTimeoutForCrawlerData: (duration) => ({ duration }),
+    fetchCrawlerData: true,
+    onCreateNewTimeout: (timeoutId) => ({ timeoutId }),
+    reApplyCrawlRules: (domain) => ({ domain }),
+    startCrawl: (overrides) => ({ overrides }),
+    stopCrawl: () => null,
+  },
+  reducers: {
+    dataLoading: [
```

Comment:
suggestion: turn this into a selector using the data in the status of GetCrawlerApiLogic, rather than a reducer using actions--slightly more robust and performant.

Reply:
The problem is I don't want it to show loading _every_ time I do a request, we make these requests on a recurring poll (~5 seconds), just on initial load.